docs/data-guidelines.md

```diff
@@ -0,0 +1,159 @@
+
+```json
+{
+  "api": {
+    "ImageData": {
+      "__compat": {},
+      "secure_context_required": {
+        "__compat": {
+          "description": "Secure context required",
+          "support": {}
+        }
+      }
+    }
+  }
+}
+```
+
+This convention is discussed in more detail in [#190](https://github.com/mdn/browser-compat-data/issues/190).
+
+
+## Web Workers (`worker_support`)
+
+Use a subfeature named `worker_support` with description text `Available in workers` to record data about an API's support for Web Workers.
+
+For example, the `ImageData` API has worker support, recorded like this:
+
+```json
+{
+  "api": {
+    "ImageData": {
+      "__compat": {},
+      "worker_support": {
+        "__compat": {
+          "description": "Available in workers",
+          "support": {}
+        }
+      }
+    }
+  }
+}
+```
+
+Formerly named `available_in_workers`, this policy was set in [#2362](https://github.com/mdn/browser-compat-data/pull/2362).
+
+
+## Non-functional defined names imply `partial_implementation`
+
+If a browser recognizes an API name, but the API doesn’t have any discernable behavior, use `"partial_implementation": true` instead of `"version_added": false`, as if the feature has non-standard support, rather than no support.
+
+For example, suppose there is some specification for a Web API `NewFeature.method()`. Running `typeof NewFeature.method` in some browser returns `function` (not `undefined`), but the method, when called, returns `null` instead of an expected value. For that feature, set `"partial_implementation": true` and write a note describing the feature’s misbehavior.
+
+See [#3904](https://github.com/mdn/browser-compat-data/pull/3904#issuecomment-484433603) for additional background (and nuance).
+
+
+## Release lines and backported features
+
+Use version numbers to reflect which _release line_ (major or minor but not patch-level releases) first supported a feature, rather than absolute version numbers.
+
+Typically, BCD does not record absolute version numbers (such as Chrome 76.0.3809.46; instead BCD records significant releases (such as Chrome 76). Use the earliest applicable release line for recording support for a given feature, even when that support change was backported to a previous release of the browser.
+
+For example, if the current release of browser X is version 10.2, but a new feature was backported to previous versions including a new 9.7.1 release, then the supported version is 9.7 (not 10.2 or 9.7.1).
+
+This decision was made in [#3953, under the expectation that most users are likely to run the latest minor version of their browser](https://github.com/mdn/browser-compat-data/pull/3953#issuecomment-485847399), but not necessarily the latest version overall.
+
+
+## Node.js releases before 1.0
```

Comment:
I'm not sure this section is needed. I had to read it twice to understand what you are referring to. Maybe it is because nodejs before 1.0 is quite rare and I wasn't dealing with this much in PRs. 

I think the core nodejs versions confusion in https://github.com/mdn/browser-compat-data/pull/3160 is that we have a "guideline" for how to do versioning with ECMAScript features (allowing only going after v8 versions) and that other nodejs versions can only be used for non-ECMAScript features (not even sure it is a guideline as we will validate this in linting in https://github.com/mdn/browser-compat-data/pull/4264). However, it is a somewhat complex situation that should probably be explained somewhere.

Reply:
Yeah, this conflates a few things, but we don't really spell any of it out anywhere. There's a few facets to this, but I think the section, as written, only addresses the third (and badly):

1. We have data for major releases and we choose not to include data for prereleases (such as Safari TPs, Microsoft Edge preview builds, etc.).
2. We make an exception for browsers' prereleases where there's a well-established release process such that the unstable versions are expected to eventually become stable (i.e., Firefox and Chrome's betas and nightlies).
3. Some Node.js releases look like prereleases (i.e., 0.10 and 0.12) but aren't really because of Node.js-specific history and that's why we allow them.
4. Some non-major version number releases of Node.js include a new version of V8 that includes an ECMAScript change and we allow those versions too.

All of these things have to do with qualitative distinctions about the data we introduce into `browsers/*.json` and I don't think the linter can help us with any of these.

I could replace this section with an overview of browser versioning in general, break these down into different sections, or not bother with any of it and chalk this up to us making judgement calls for each of the releases, as they're added.

Reply:
I think all 4 points need some more thoughts. I would vote to remove the section for now and file this discussion as a follow-up. It's worth digging into some of these things on their own in more details.

Reply:
OK, thanks. I've dropped the section from this PR. And opened #4492.